### ISSUE TITLE
Understanding how to use LBFGS

### ISSUE THREAD

**Issue by juicyslew** [brendano/myutil#2]
Dear Mr. O'Connor,

I am a high school senior transitioning to college (graduated yesterday). I am trying to implement LBFGS into an application I am making. I understand the basic math behind Neural Networks and how to perform backpropagation and receive the error for all the weights. I am currently working with a friend who is handling the Java and organization part of the application, while I am doing the Neural Network and Mathy part of it. I have been reading through this code, but as I don't understand how to use Java very well yet, and dont understand the actual algorithm of LBFGS, I am at a loss of how to use my Original data, cost function, weights, and backpropagation error of these weights in your function in order to produce results that allow for minimizing the cost function and creating a good predicting device for what I wish to predict. If you could shine light on how to do so, I would be very grateful.

TL;DR: How do I use my original data, cost function, weights of a neural network, and error of these weights to find an optimization of the neural network using your LBFGS function. Where do I put in these parameters and in what format?

I know this is kinda a stupid question and will probably boil down to a simple answer, but I am very lost with the code and how to use it.

A student,
William Derksen

**Comment by brendano** [comment]
Hello!  Sorry, we don't have any documentation for what you want and it would be a large effort to write such documentation.  You should look for a tutorial on learning neural networks with backpropagation that illustrates the entire system, and ideally demonstration code you can build off of.  LBFGS is only one part of the story (and there are alternatives for that part anyway, like stochastic gradient descent).

**Comment by juicyslew** [comment]
Thank you I will check out that alternative. Are there any other alternatives that you would suggest looking in to?

**Comment by brendano** [comment]
No, just try to find tutorials on the subject that make sense to you.  There are many out there these days.
